src/endpoints/authentication/impl.cpp

```diff
@@ -82,33 +83,47 @@ namespace irods::http::handler
             bool login_successful = false;
 
             try {
-                static const auto& rodsadmin_username = irods::http::globals::configuration().at(nlohmann::json::json_pointer{"/irods_client/proxy_admin_account/username"}).get_ref<const std::string&>();
-                static const auto& rodsadmin_password = irods::http::globals::configuration().at(nlohmann::json::json_pointer{"/irods_client/proxy_admin_account/password"}).get_ref<const std::string&>();
-                static const auto& zone = irods::http::globals::configuration().at(nlohmann::json::json_pointer{"/irods_client/zone"}).get_ref<const std::string&>();
+                using json_pointer = nlohmann::json::json_pointer;
 
-                CheckAuthCredentialsInput input{};
-                username.copy(input.username, sizeof(CheckAuthCredentialsInput::username));
-                zone.copy(input.zone, sizeof(CheckAuthCredentialsInput::zone));
+                static const auto& config = irods::http::globals::configuration();
+                static const auto& rodsadmin_username = config.at(json_pointer{"/irods_client/proxy_admin_account/username"}).get_ref<const std::string&>();
+                static const auto& rodsadmin_password = config.at(json_pointer{"/irods_client/proxy_admin_account/password"}).get_ref<const std::string&>();
+                static const auto& zone = config.at(json_pointer{"/irods_client/zone"}).get_ref<const std::string&>();
 
-                namespace adm = irods::experimental::administration;
-                const adm::user_password_property prop{password, rodsadmin_password};
-                const auto obfuscated_password = irods::experimental::administration::obfuscate_password(prop);
-                obfuscated_password.copy(input.password, sizeof(CheckAuthCredentialsInput::password));
+                if (config.at(json_pointer{"/irods_client/enable_4_2_compatibility"}).get<bool>()) {
```

Comment:
add a couple comments on what we have to  / will do differently in this block.

Reply:
done